mobile-sdk/flowcharts.md

```diff
@@ -0,0 +1,320 @@
+            WebView ->> SDK: Navigate to paymentUrl ②
+            SDK ->> SDK: Recognize paymentUrl
+            SDK ->> WebView: Cancel navigation
+        else Not Compatible with Web Wiew
+            SDK ->> WebView: Cancel navigation
+            SDK ->> System: Open URL
+            System ->> Browser: Open URL in Browser
+            User ->> Browser: Handle external flow
+            Browser ->> User: Handle external flow
+            Browser -> Browser: Open paymentUrl in Browser
+        end
+    else Navigation is to an app-specific URL (custom scheme, Android Deep Link/App Link, iOS Universal Link)
+        SDK ->> WebView: Cancel navigation
+        SDK ->> System: Open URL
+        System ->> Ext: Launch URL-appropriate app
+        User ->> Ext: Handle external flow
+        Ext ->> User: Handle external flow
+        Ext ->> System: Open paymentUrl
+        alt Android
+            System ->> Browser: Open paymentUrl in Browser ③
+        else iOS
+            System ->> App: Launch app with paymentUrl ④
+        end
+    end
+```
+
+*   ① The same check is repeated for any further navigation inside the WebView
+*   ② All properly configured authentication flows must end up here
+*   ③ On Android, paymentUrl is an https URL that redirects to an Android Intent URL.
+*   ④ On iOS, paymentUrl is a Universal Link. When an app open a Universal Link to\nanother app, it should be routed to that app instead of the Browser. However, Univeral Links are finicky things, and it is not impossible that it gets opened in the Browser instead. In that case, the flow continues with "paymentUrl opened in Browser" below instead.
+
+### Return from Browser
+
+If the external flow ended with `paymentUrl` opened in the browser, we need a way to get back to the app. On Android, this is simple to accomplish by redirecting to an [Android Intent Uri][android-intent-scheme]; the SDK and backend work together to construct the Intent Uri to point to the correct app. This Intent will cause the app to be brought back into focus, and the PaymentFragment will recognize the `paymentUrl` and reload the payment menu.
+
+On iOS, the situation is more complicated. As mentioned above, `paymentUrl` is a Universal Link, and navigation to it should be routed to the app. However, Universal Links are a bit unreliable, in that they require certain conditions to be fulfilled; otherwise, they are opened in the browser like regular links. Unfortunately, one of the conditions, namely that the navigation originated from the user pressing on a link, is often not fulfilled in the external pages used by payment methods. Therefore, we must have a system that works correctly, even if `paymentUrl` is opened in the browser.
+
+On iOS, we use the following system:
+ - `paymentUrl` redirects (301) to a trampoline page hosted at a different domain
+ - the trampoline page has a button
+ - pressing that button navigates to `paymentUrl` but with an extra parameter
```

Comment:
Codacy found an issue: [[unordered-list-marker-style] Marker style should be `*`](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5758572)